homeassistant/components/tessie/config_flow.py

```diff
@@ -0,0 +1,104 @@
+"""Config Flow for Tessie integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from http import HTTPStatus
+from typing import Any
+
+from aiohttp import ClientConnectionError, ClientResponseError
+from tessie_api import get_state_of_all_vehicles
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ACCESS_TOKEN
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+TESSIE_SCHEMA = vol.Schema({vol.Required(CONF_ACCESS_TOKEN): str})
+
+
+class TessieConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config Tessie API connection."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize."""
+        self._reauth_entry: ConfigEntry | None = None
+
+    async def async_step_user(
+        self, user_input: Mapping[str, Any] | None = None
+    ) -> FlowResult:
+        """Get configuration from the user."""
+        errors = {}
+        if user_input and CONF_ACCESS_TOKEN in user_input:
+            try:
+                await get_state_of_all_vehicles(
+                    session=async_get_clientsession(self.hass),
+                    api_key=user_input[CONF_ACCESS_TOKEN],
+                )
+            except ClientResponseError as e:
+                if e.status == HTTPStatus.UNAUTHORIZED:
+                    errors["base"] = "invalid_access_token"
+                else:
+                    errors["base"] = "unknown"
+            except ClientConnectionError:
+                errors["base"] = "cannot_connect"
+            else:
+                await self.async_set_unique_id(user_input[CONF_ACCESS_TOKEN])
```

Comment:
It looks like reauth will be a problem as the unique id should not change for the life of the config entry

Is there a better unique id available such as the account id ?

Reply:
Hrm, thats an interesting problem. As it stands no there is not a better unique that is static. I have asked the Tessie developer if they can add an API endpoint that returns the users email address, but its not there yet.

Can I instead force there to only ever one be instance of this integration? For example setting the unique id to "tessie". Realistically no one should have **two** paid subscriptions to this service, since you can add multiple cars in a single subscription.

Reply:
Single entry only would be fine as you can always update it later.

In this case we would simply not set a unique id at all

Reply:
I have removed the unique id until I can get the users email address.